internal/common/compose_result.go

```diff
@@ -6,32 +6,33 @@ import (
 	"io"
 )
 
+type Assembler struct {
+	Name    string          `json:"name"`
+	Options json.RawMessage `json:"options"`
+	Success bool            `json:"success"`
+	Output  string          `json:"output"`
+}
+
+type Stages []struct {
```

Comment:
I think I prefer having non-array `Stage` type, it's more common to define non-array types.

Reply:
Alright, I'm still kinda confused about these types. I changed it to hopefully a better state.